Harder to install 0.16.0 due to dependency on brotli-size

Hi! Thank you for a great tool :-)

**Do you want to request a *feature* or report a *bug*?**
Ease of use regression between 0.15.3 and 0.16.0.

**What is the current behavior?**
#194 (released in 0.16.0) added support for tracking the brotli sizes of files, however to do so it had to add a dependency on [brotli-size](https://www.npmjs.com/package/brotli-size), which itself has a dependency on [iltorb](https://www.npmjs.com/package/iltorb). 

If iltorb isn't able to find a pre-compiled binary, it has to build using node-gyp, which requires Python and g++ be installed - which often aren't in Docker images, or in Windows environments.

For users that aren't making use of the brotli size feature, it would be a shame to have to install Python/g++ just to allow bundlesize to install. Perhaps `brotli-size` could be made an optional dependency?

It's worth also noting that:
* compiled dependencies are pretty painful at the moment even when they succeed, due to yarnpkg/yarn#932
* the `brotli-size` dependency doubled the install size and package dependency count of bundlesize (taking it up to ~200 dependencies!)
* iltorb currently doesn't work with repositories that are using the `no-bin-links` feature (necessary when using npm/yarn inside a linux Vagrant environment using shared folders with a Windows host). I'll be filing an issue against iltorb shortly for this. _Edit: filed MayhemYDG/iltorb#58_

**If the current behavior is a bug, please provide the steps to reproduce.**
`docker run --rm -it node:9-slim yarn add bundlesize@0.16.0 --dev`
...which results in: https://emorley.pastebin.mozilla.org/9078461

**What is the expected behavior?**
Successful install, like with 0.15.3:
`docker run --rm -it node:9-slim yarn add bundlesize@0.15.3 --dev`
...which results in: https://emorley.pastebin.mozilla.org/9078471

**Please mention other relevant information.**
- node version: 9.6.0
- npm version: N/A (yarn 1.3.2)
- Operating system: Multiple (but Debian jessie for the example above)
- bundlesize version: 0.16.0

> **If the current behavior is a bug, please provide the steps to reproduce.**
> `docker run --rm -it node:9-slim yarn add bundlesize@0.16.0 --dev`
> ...which results in: https://emorley.pastebin.mozilla.org/9078461
> 
> **What is the expected behavior?**
> Successful install, like with 0.15.3:
> `docker run --rm -it node:9-slim yarn add bundlesize@0.15.3 --dev`
> ...which results in: https://emorley.pastebin.mozilla.org/9078471

These pastebin links are 404ing right now. Here is a 2019 update:

**If the current behavior is a bug, please provide the steps to reproduce.**
`docker run --rm -it node:12-slim npx bundlesize@0.16.0`
...which results in: 

<details>
<summary>The following log</summary>

```
sh: 1: detect-libc: Permission denied
gyp ERR! configure error 
gyp ERR! stack Error: Can't find Python executable "python", you can set the PYTHON env variable.
gyp ERR! stack     at PythonFinder.failNoPython (/usr/local/lib/node_modules/npm/node_modules/node-gyp/lib/configure.js:484:19)
gyp ERR! stack     at PythonFinder.<anonymous> (/usr/local/lib/node_modules/npm/node_modules/node-gyp/lib/configure.js:406:16)
gyp ERR! stack     at F (/usr/local/lib/node_modules/npm/node_modules/which/which.js:68:16)
gyp ERR! stack     at E (/usr/local/lib/node_modules/npm/node_modules/which/which.js:80:29)
gyp ERR! stack     at /usr/local/lib/node_modules/npm/node_modules/which/which.js:89:16
gyp ERR! stack     at /usr/local/lib/node_modules/npm/node_modules/isexe/index.js:42:5
gyp ERR! stack     at /usr/local/lib/node_modules/npm/node_modules/isexe/mode.js:8:5
gyp ERR! stack     at FSReqCallback.oncomplete (fs.js:149:21)
gyp ERR! System Linux 5.0.8-arch1-1-ARCH
gyp ERR! command "/usr/local/bin/node" "/usr/local/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd /root/.npm/_npx/1/lib/node_modules/bundlesize/node_modules/iltorb
gyp ERR! node -v v12.3.1
gyp ERR! node-gyp -v v3.8.0
gyp ERR! not ok 
[############......] | install: info lifecycle iltorb@1.3.10~install: Failed to
[############......] | install: info lifecycle iltorb@1.3.10~install: Failed to
[############......] | install: info lifecycle iltorb@1.3.10~install: Failed to

npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! iltorb@1.3.10 install: `detect-libc prebuild-install || node-gyp rebuild`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the iltorb@1.3.10 install script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /root/.npm/_logs/2019-05-24T16_21_06_822Z-debug.log
Install for [ 'bundlesize@0.16.0' ] failed with code 1

```

</details>

**What is the expected behavior?**
Successful install, like with 0.15.3

--- a/.travis.yml
+++ b/.travis.yml
@@ -1,7 +1,7 @@
 language: node_js
 node_js:
-  - 'node' # latest: 12
-  - 'lts' # LTS: 10
+  - 12 # Latest
+  - 10 # LTS
   - 8
 cache:
   directories:

--- a/.travis.yml
+++ b/.travis.yml
@@ -1,5 +1,7 @@
 language: node_js
 node_js:
+  - 'node' # latest: 12
+  - 'lts' # LTS: 10
   - 8
 cache:
   directories:

--- a/package.json
+++ b/package.json
@@ -37,10 +37,10 @@
   "license": "MIT",
   "dependencies": {
     "axios": "^0.19.0",
-    "brotli-size": "0.1.0",
     "bytes": "^3.1.0",
     "ci-env": "^1.4.0",
     "commander": "^2.20.0",
+    "compare-versions": "^3.5.0",
     "cosmiconfig": "^5.2.1",
     "github-build": "^1.2.0",
     "glob": "^7.1.4",
@@ -64,7 +64,8 @@
     "babel-preset-es2015": "^7.0.0-beta.3",
     "babel-preset-stage-3": "^7.0.0-beta.3",
     "babel-traverse": "^7.0.0-beta.3",
-    "execa": "^2.0.1",
+    "brotli-size": "^0.1.0",
+    "execa": "^2.0.2",
     "husky": "^0.14.3",
     "lint-staged": "^7.1.1",
     "prettier": "^1.6.0"

--- a/package.json
+++ b/package.json
@@ -40,7 +40,6 @@
     "bytes": "^3.1.0",
     "ci-env": "^1.4.0",
     "commander": "^2.20.0",
-    "compare-versions": "^3.5.0",
     "cosmiconfig": "^5.2.1",
     "github-build": "^1.2.0",
     "glob": "^7.1.4",

--- a/package.json
+++ b/package.json
@@ -64,7 +64,7 @@
     "babel-preset-es2015": "^7.0.0-beta.3",
     "babel-preset-stage-3": "^7.0.0-beta.3",
     "babel-traverse": "^7.0.0-beta.3",
-    "execa": "^2.0.2",
+    "execa": "^2.0.1",
     "husky": "^0.14.3",
     "lint-staged": "^7.1.1",
     "prettier": "^1.6.0"

--- a/package.json
+++ b/package.json
@@ -64,7 +64,6 @@
     "babel-preset-es2015": "^7.0.0-beta.3",
     "babel-preset-stage-3": "^7.0.0-beta.3",
     "babel-traverse": "^7.0.0-beta.3",
-    "brotli-size": "^0.1.0",
     "execa": "^2.0.2",
     "husky": "^0.14.3",
     "lint-staged": "^7.1.1",

--- a/package.json
+++ b/package.json
@@ -63,6 +63,7 @@
     "babel-preset-es2015": "^7.0.0-beta.3",
     "babel-preset-stage-3": "^7.0.0-beta.3",
     "babel-traverse": "^7.0.0-beta.3",
+    "bundlesize-plugin-brotli": "^1.0.0",
     "execa": "^2.0.1",
     "husky": "^0.14.3",
     "lint-staged": "^7.1.1",

--- a/package.json
+++ b/package.json
@@ -63,7 +63,6 @@
     "babel-preset-es2015": "^7.0.0-beta.3",
     "babel-preset-stage-3": "^7.0.0-beta.3",
     "babel-traverse": "^7.0.0-beta.3",
-    "bundlesize-plugin-brotli": "^1.0.0",
     "execa": "^2.0.1",
     "husky": "^0.14.3",
     "lint-staged": "^7.1.1",

--- a/packages/bundlesize-plugin-brotli/index.js
+++ b/packages/bundlesize-plugin-brotli/index.js
@@ -1,0 +1,3 @@
+const brotliSize = require('brotli-size')
+
+module.exports = { sync: brotliSize.sync }

--- a/packages/bundlesize-plugin-brotli/package.json
+++ b/packages/bundlesize-plugin-brotli/package.json
@@ -1,0 +1,12 @@
+{
+  "name": "bundlesize-plugin-brotli",
+  "version": "1.0.0",
+  "description": "Plugin to use brotli compression with bundlesize",
+  "main": "index.js",
+  "keywords": [],
+  "author": "",
+  "license": "MIT",
+  "dependencies": {
+    "brotli-size": "0.1.0"
+  }
+}

--- a/packages/bundlesize-plugin-brotli/readme.md
+++ b/packages/bundlesize-plugin-brotli/readme.md
@@ -1,0 +1,56 @@
+<p align="center">
+  <img src="https://cdn.rawgit.com/siddharthkp/bundlesize/master/art/logo.png" height="200px">
+  <br><br>
+  <b>Plugin to use brotli compression with bundlesize on Node < 10.6.0</b>
+  <br>
+</p>
+
+&nbsp;
+
+#### Note:
+
+If you are using Node version >= 10.16.0, you do not need this plugin.
+
+&nbsp;
+
+#### Install
+
+```sh
+npm install bundlesize-plugin-brotli --save-dev
+
+# or
+
+yarn add bundlesize-plugin-brotli --dev
+```
+
+&nbsp;
+
+#### Setting up bundlesize
+
+&nbsp;
+
+See bundlesize usage here: https://github.com/siddharthkp/bundlesize
+
+&nbsp;
+
+#### Using brotli compression
+
+&nbsp;
+
+By default, bundlesize `gzips` your build files before comparing.
+
+If you are using `brotli` instead of gzip, you can specify that with each file:
+
+```json
+{
+  "files": [
+    {
+      "path": "./build/vendor.js",
+      "maxSize": "5 kB",
+      "compression": "brotli"
+    }
+  ]
+}
+```
+
+&nbsp;

--- a/src/brotli.js
+++ b/src/brotli.js
@@ -1,0 +1,43 @@
+const compare = require('compare-versions')
+const { error } = require('prettycli')
+const zlib = require('zlib')
+
+const config = require('./config')
+
+function getBrotliSync() {
+  // Does this project want brotli compression?
+  const needsBrotli = config.find(row => row.compression === 'brotli')
+
+  // If it doesn't, save us the trouble.
+  if (!needsBrotli) return null
+
+  // Check if the installed version of node supports brotli
+  const hasBrotli = zlib.brotliCompressSync
+  if (hasBrotli) return nativeBrotliSync
+
+  // Looks like this version of node does not have brotli
+  // We recommend using bundlesize-plugin-brotli which acts
+  // like a polyfill
+
+  try {
+    const polyfill = require('bundlesize-plugin-brotli')
+    // if the user has installed the plugin, we can safely return it
+    return polyfill
+  } catch (err) {
+    // if they haven't, show them an error and exit with error code 1
+    const message = `Missing dependency: bundlesize-plugin-brotli
+
+  To use compression: brotli, please install bundlesize-plugin-brotli as a dev dependency.
+
+  You can read about the compression options here:
+  https://github.com/siddharthkp/bundlesize#customisation`
+
+    error(message, { silent: true })
+  }
+}
+
+function nativeBrotliSync(input) {
+  return zlib.brotliCompressSync(input).length
+}
+
+module.exports = { sync: getBrotliSync() }

--- a/src/brotli.js
+++ b/src/brotli.js
@@ -21,7 +21,7 @@ function getBrotliSync() {
   try {
     const polyfill = require('bundlesize-plugin-brotli')
     // if the user has installed the plugin, we can safely return it
-    return polyfill
+    return polyfill.sync
   } catch (err) {
     // if they haven't, show them an error and exit with error code 1
     if (err && err.code === 'MODULE_NOT_FOUND') {

--- a/src/brotli.js
+++ b/src/brotli.js
@@ -1,4 +1,3 @@
-const compare = require('compare-versions')
 const { error } = require('prettycli')
 const zlib = require('zlib')
 

--- a/src/brotli.js
+++ b/src/brotli.js
@@ -24,7 +24,8 @@ function getBrotliSync() {
     return polyfill
   } catch (err) {
     // if they haven't, show them an error and exit with error code 1
-    const message = `Missing dependency: bundlesize-plugin-brotli
+    if (err && err.code === 'MODULE_NOT_FOUND') {
+      const message = `Missing dependency: bundlesize-plugin-brotli
 
   To use brotli with Node versions lower than v10.16.0,
   please install bundlesize-plugin-brotli as a dev dependency.
@@ -32,7 +33,11 @@ function getBrotliSync() {
   You can read about the compression options here:
   https://github.com/siddharthkp/bundlesize#customisation`
 
-    error(message, { silent: true })
+      error(message, { silent: true })
+    } else {
+      // if it's a different error, show the raw error
+      error(err, { silent: true })
+    }
   }
 }
 

--- a/src/brotli.js
+++ b/src/brotli.js
@@ -27,7 +27,8 @@ function getBrotliSync() {
     // if they haven't, show them an error and exit with error code 1
     const message = `Missing dependency: bundlesize-plugin-brotli
 
-  To use compression: brotli, please install bundlesize-plugin-brotli as a dev dependency.
+  To use brotli with Node versions lower than v10.16.0,
+  please install bundlesize-plugin-brotli as a dev dependency.
 
   You can read about the compression options here:
   https://github.com/siddharthkp/bundlesize#customisation`

--- a/src/compressed-size.js
+++ b/src/compressed-size.js
@@ -1,5 +1,5 @@
 const gzip = require('gzip-size')
-const brotli = require('brotli-size')
+const brotli = require('./brotli')
 
 const getCompressedSize = (data, compression = 'gzip') => {
   let size

